src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -208,11 +209,72 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
         public static string LookupTargetFrameworkFromProjectFile(string projectLocation)
         {
             var projectFile = FindProjectFileInDirectory(projectLocation);
+            if (string.IsNullOrEmpty(projectFile))
+            {
+                throw new FileNotFoundException("Could not find a project file in the specified directory.");
+            }
 
-            var xdoc = XDocument.Load(projectFile);
+            var arguments = new[]
+            {
+                "msbuild",
+                projectFile,
+                "--getProperty:TargetFramework,TargetFrameworks",
+                "-nologo"
+            };
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = string.Join(" ", arguments),
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                }
+            };
+
+            process.Start();
+            string output = process.StandardOutput.ReadToEnd();
+            string error = process.StandardError.ReadToEnd();
+            process.WaitForExit();
+
+            if (process.ExitCode != 0)
```

Comment:
If the process failed we should fallback to the original XML parsing logic. There could be CI systems that are still deploying .NET 6 and only have .NET 6 SDK installed. The `-getoutput` command was added in .NET 8.

Reply:
ah good point! i will update accordingly.